File: Calendar/wwwroot/js/models/Event.js
import { EventRepository } from '../models/mvc/event-repository.js';

export class Event {
  constructor(title, description, start, finish, color, isAllDay, notify) {
    this.data = {
      Title: title,
      Description: description,
      Start: start,
      Finish: finish,
      Color: color,
      IsAllDay: isAllDay,
      Notify: notify
    };
  }
  
  sendToMVC() {
    new EventRepository().insert(this.data);
  }
}

File: Data/Models/Event.cs
namespace Data.Models
{
    using System;

    public class Event
    {
        public int Id { get; set; }
        public int CalendarId { get; set; }
        public string Description { get; set; }
        public string Title { get; set; }
        public DateTime TimeStart { get; set; }
        public DateTime TimeFinish { get; set; }
        public bool AllDay { get; set; }
        public int RepeatId { get; set; }
    }
}